src/loader/mod.rs

```diff
@@ -271,7 +280,20 @@ mod tests {
         assert_eq!(val, b'3');
         cmdline_address = cmdline_address.unchecked_add(1);
         let val: u8 = gm.read_obj(cmdline_address).unwrap();
-        assert_eq!(val, b'4');
+        assert_eq!(val, b'\0');
+
+        // Overwrite guest's memory where cmdline was previously written with a shorter
+        // cmdline to check that the null terminator is written in guest's memory
+        let mut cl = Cmdline::new(10);
+        cl.insert_str("12").unwrap();
+        assert_eq!(cl.as_cstring().unwrap().as_bytes_with_nul(), &[49, 50, 0]);
+        assert_eq!(Ok(()), load_cmdline(&gm, cmdline_address, &cl));
```

Comment:
```suggestion
        load_cmdline(&gm, cmdline_address, &cl).unwrap();
```

Reply:
yes, doing this way; replicating for the first write of the cmdline into memory; if the load_cmdline() fails, the unwrap() will panic so the test fails.